.github/workflows/validate-kustomize.yml

```diff
@@ -0,0 +1,68 @@
+name: Validate manifests with Kustomize
+
+on:
+  push:
+    paths:
+      - 'manifests/**'
+  pull_request:
+    paths:
+      - 'manifests/**'
+  workflow_dispatch: {}
+
+concurrency:
+  group: kustomize-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  kustomize-build:
+    name: Build ${{ matrix.target.name }}
+    runs-on: ubuntu-latest
+    env:
+      # Pin the kustomize version and expose it to steps and cache key
+      KUSTOMIZE_VERSION: v5.4.1
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - name: RHOAI Add-on
+            path: manifests/rhoai/addon
+          - name: RHOAI On-Prem
+            path: manifests/rhoai/onprem
+          - name: ODH
+            path: manifests/odh
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
```

Comment:
Perhaps do a sparse checkout of the manifests dir only?

Reply:
Yes, adding a sparse checkout.